### PR TITLE
[ElmSharp] Added some new APIs for AnimationView

### PR DESCRIPTION
**Pull request by arosis78**
### Description of Change ###
Added some new APIs for AnimationView

### API Changes ###
Changed:
AnimationView.KeyFrame -> AnimationView.Progress

Added:
public enum AnimationViewState
AnimationView.Frame { get; set; } // Property
AnimationView.DefaultSize { get; } // Property
AnimationView.State { get; } // Property
AnimationView.IsReversePlaying { get; } // Property
AnimationView.FrameCount { get; } // Property
AnimationView.MinProgress { get; set; } // Property
AnimationView.MaxProgress { get; set; } // Property
AnimationView.MinFrame { get; set; } // Property
AnimationView.MaxFrame { get; set; } // Property

### CODE REVIEW

File: src/ElmSharp/ElmSharp/AnimationView.cs [hunk 1]
```diff
@@ -207,23 +239,173 @@ public double DurationTime
...
+        /// Get the status whether current animation is on playing forward or backward.
+        /// </summary>
         /// <remarks>
-        /// When you required to jump on a certain frame instantly,
-        /// you can change current keyframe by using this property
-        /// The range of keyframe is 0 ~ 1.
+        /// If AnimationView is not on playing, it will return False.
         /// </remarks>
+        /// <since_tizen> preview </since_tizen>
+        public bool IsReversePlaying
```
Comment (myroot) [inline src/ElmSharp/ElmSharp/AnimationView.cs]:
```suggestion
        public bool IsReversedPlaying
```

File: src/ElmSharp/ElmSharp/AnimationView.cs [hunk 2]
```diff
@@ -207,23 +239,173 @@ public double DurationTime
...
+        }
+
+        /// <summary>
+        /// Sets or Gets the start frame of the play
+        /// </summary>
+        /// <remarks>
+        /// Default value is 0.
+        /// </remarks>
+        /// <since_tizen> preview </since_tizen>
+        public int MinFrame
```
Comment (myroot) [inline src/ElmSharp/ElmSharp/AnimationView.cs]:
```suggestion
        public int FirstFrame
```

File: src/ElmSharp/ElmSharp/AnimationView.cs [hunk 3]
```diff
@@ -207,23 +239,173 @@ public double DurationTime
...
+
+        /// <summary>
+        /// Sets or Gets the last frame of the play
+        /// </summary>
+        /// <remarks>
+        /// Default value is FrameCount -1.
+        /// </remarks>
         /// <since_tizen> preview </since_tizen>
-        public double KeyFrame
+        public int MaxFrame
```
Comment (myroot) [inline src/ElmSharp/ElmSharp/AnimationView.cs]:
```suggestion
        public int LastFrame
```

File: src/ElmSharp/ElmSharp/AnimationView.cs (hunk 1 above)
Comment (arosis78) [reply src/ElmSharp/ElmSharp/AnimationView.cs]: Done

File: src/ElmSharp/ElmSharp/AnimationView.cs (hunk 2 above)
Comment (arosis78) [reply src/ElmSharp/ElmSharp/AnimationView.cs]: Lottie is provided APIs for play animation segments such as setMinFrame(), setMinProgress(), and so on. I know that EFL native animation view is implemented according to this lottie lib APIs.
Please refer to page (https://airbnb.io/lottie/android/android.html#sample-app)
I think ElmSharp should follow this APIs name like native EFL and lottie.

File: src/ElmSharp/ElmSharp/AnimationView.cs (hunk 3 above)
Comment (arosis78) [reply src/ElmSharp/ElmSharp/AnimationView.cs]: Lottie is provided APIs for play animation segments such as setMinFrame(), setMinProgress(), and so on. I know that EFL native animation view is implemented according to this lottie lib APIs.
Please refer to page (https://airbnb.io/lottie/android/android.html#sample-app)
I think ElmSharp should follow this APIs name like native EFL and lottie.

File: src/ElmSharp/ElmSharp/AnimationView.cs [hunk 4]
```diff
@@ -18,6 +18,38 @@
...
+
+        /// <summary>
+        /// Animation is on playing.
+        /// </summary>
+        Play,
+
+        /// <summary>
+        /// Animation is on reverse playing.
+        /// </summary>
+        ReversePlay,
```
Comment (myroot) [inline src/ElmSharp/ElmSharp/AnimationView.cs]:
```suggestion
        ReversedPlay,
```

**Pull request merged**